src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -580,6 +580,7 @@ public void Dispose_UseAfterDispose_Throws()
             Assert.Throws<ObjectDisposedException>(() => { client.PostAsync(CreateFakeUri(), new ByteArrayContent(new byte[1])); });
             Assert.Throws<ObjectDisposedException>(() => { client.PutAsync(CreateFakeUri(), new ByteArrayContent(new byte[1])); });
             Assert.Throws<ObjectDisposedException>(() => { client.SendAsync(new HttpRequestMessage(HttpMethod.Get, CreateFakeUri())); });
+            Assert.Throws<ObjectDisposedException>(() => { client.Send(new HttpRequestMessage(HttpMethod.Get, CreateFakeUri())); });
             Assert.Throws<ObjectDisposedException>(() => { client.Timeout = TimeSpan.FromSeconds(1); });
         }
```

Comment:
We need some sync-specific tests, for things like validating that, to whatever extent we can test, all portions of the processing are happening on the same thread synchronously.  For example, plug in a custom HttpContent for request content and assert that it's called on the same thread that's calling Send.  We also need to ensure all of our tests related to cancellation are appropriately running for the sync entrypoint, and potentially beef those up even further; that's an area where we're likely to find some issues with the sync APIs and it'd be good to shake out as much of that ahead of time.

Reply:
Added tests.